pkg/cluster/controllers/harbor/harbor.go

```diff
@@ -117,7 +114,36 @@ func (harbor *Controller) getHarborCR(ctx context.Context, harborcluster *goharb
 				*metav1.NewControllerRef(harborcluster, goharborv1.HarborClusterGVK),
 			},
 		},
-		Spec: spec,
+		Spec: goharborv1.HarborSpec{
+			ExternalURL: spec.ExternalURL,
+			InternalTLS: goharborv1.HarborInternalTLSSpec{
+				Enabled: spec.InternalTLS.Enabled,
+			},
+			ImageChartStorage:      &goharborv1.HarborStorageImageChartStorageSpec{},
+			LogLevel:               spec.LogLevel,
+			HarborAdminPasswordRef: spec.HarborAdminPasswordRef,
+			UpdateStrategyType:     spec.UpdateStrategyType,
+			Version:                spec.Version,
+			Expose:                 spec.Expose,
+			HarborComponentsSpec:   spec.HarborComponentsSpec,
+			ImageSource:            spec.ImageSource,
+			Proxy:                  spec.Proxy,
+		},
+	}
+
+	if harborcluster.Spec.Storage.Spec.FileSystem != nil {
```

Comment:
We need to make sure only one storage is defined in the spec via webhook?

Reply:
If there is already a Kind distinction, it looks like it should not be needed

Reply:
As I mentioned, the storage definitions are marked optional with pointers. But the user can also provide more than 1 storage in the spec. e.g:

```
spec:
  kind: s3
  spec:
    filesystem: {},
    s3: {},
    minio: {}
```

It seems we did not use spec.kind in the spec. If we did not add validating logic in the validating webhook, we need to take care of code here. Providing validating webhook is more simple, right?

@cndoit18

Reply:
or:

```yaml
spec:
  kind: s3
 spec:
   minio:
    size: 100GiB
```